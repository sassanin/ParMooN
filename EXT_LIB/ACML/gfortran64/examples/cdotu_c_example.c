
/* cdotu Example Program Text */
/*
 * ACML version 1.0 Copyright AMD,NAG 2003
 */

#include <acml.h>
#include <stdio.h>

int main(void)
{
  complex r;
  complex x[3], y[3];
  int n = 3;
  int i;

  printf("ACML example: dot product of two complex vectors using cdotu\n");
  printf("------------------------------------------------------------\n");
  printf("\n");

  x[0] = compose_complex(1.0, 2.0);
  x[1] = compose_complex(2.0, 1.0);
  x[2] = compose_complex(1.0, 3.0);
  y[0] = compose_complex(3.0, 1.0);
  y[1] = compose_complex(1.0, 4.0);
  y[2] = compose_complex(1.0, 2.0);

  printf("Vector x: ");
  for (i = 0; i < n; i++)
    printf(" (%7.4f,%7.4f)\n", x[i].real, x[i].imag);
  printf("Vector y: ");
  for (i = 0; i < n; i++)
    printf(" (%7.4f,%7.4f)\n", y[i].real, y[i].imag);

  r = cdotu(n, x, 1, y, 1);

  printf("r = x.y = (%12.3f,%12.3f)\n", r.real, r.imag);
  return 0;
}
