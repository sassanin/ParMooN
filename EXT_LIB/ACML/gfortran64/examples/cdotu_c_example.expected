ACML example: dot product of two complex vectors using cdotu
------------------------------------------------------------

Vector x:  ( 1.0000, 2.0000)
 ( 2.0000, 1.0000)
 ( 1.0000, 3.0000)
Vector y:  ( 3.0000, 1.0000)
 ( 1.0000, 4.0000)
 ( 1.0000, 2.0000)
r = x.y = (      -6.000,      21.000)
